startmenu.py

```diff
@@ -63,16 +66,82 @@ def print_menu(screen, selected_row_index):
     screen.refresh()
 
 
-def start_game(screen):
+
+
+
+def validate_key_input(key_input):
+    """
+    Callback function used by curses when a user types input.
+    The function checks if the pressed key is one of the enter keys and
+    signals curses to stop asking for input. Otherwise it lets the character through
+    """
+    if key_input in enter_keys:
+        # 7 is a magic number that tells curses to stop asking for input
+        return 7
+    else:
+        return key_input
+
+
+def get_player_name(screen, text):
+    """
+    Prints and centers text on screen.
+    Creates a new text input where the user enters the player name and returns it.
+
+    Keyword arguments:
+    screen             -- the curses screen.
+    text               -- Text that appears before the input.
+
+    Example:
+    text = "Insert player 1's name" => Insert player 1's name: (user types here)
+
+    Returns:
+    The name of the player that the user entered.
+    """
+
+    # Centers the text
+    num_rows, num_cols = screen.getmaxyx()
+
+    x = int(num_cols / 2) - int(len(text) / 2)
+    y = int(num_rows / 2)
+
+    screen.addstr(y, x, text)
+    screen.refresh()
+
+    # We must create a new window becuase the edit function will return
+    # everything that has been printed on the screen and not just the entered name
+    win = curses.newwin(5, 10, y, x + len(text))
+    textbox = Textbox(win)
+    player_name = textbox.edit(validate_key_input)
+
+    return player_name
+
+def startgame(screen, player1_name, player2_name):
+    screen.clear()
+    screen.addstr(0, 0, "Player1: " + player1_name)
+    screen.addstr(1, 0, "Player2: " + player2_name)
+    screen.refresh()
+
+    time.sleep(1)
+
+def ask_for_player_names(screen):
     """Temporary functions that 'starts' the game.
 
     Keyword arguments:
     screen -- the curses screen
     """
-    screen.addstr(0, 0, "Game will now start")
-    screen.refresh()
+    player1_text = "Inset player 1's name: "
+    player2_text = "Insert player 2's name: "
+
+    # Enable blinking cursor when typing in names
+    curses.curs_set(1)
+
+    player1_name = get_player_name(screen, player1_text)
+    player2_name = get_player_name(screen, player2_text)
+
+    curses.curs_set(0)
+    #instead of startgame()
+    game_board.main(screen,player1_name,player2_name)
```

Comment:
Move the call to `game_board.main` to main. Put it just after you call `ask_for_player_names`. Starting the game and asking for the players names are two different things and should therefor be separated